pkg/kubelet/cm/deviceplugin/device_plugin_handler.go

```diff
@@ -226,6 +244,57 @@ func (h *HandlerImpl) Allocate(pod *v1.Pod, container *v1.Container, activePods
 	return h.writeCheckpoint()
 }
 
+// Allocate attempts to allocate all of required device plugin resources,
+// and update Allocatable resources in nodeInfo if necessary
+func (h *HandlerImpl) Allocate(node *schedulercache.NodeInfo, attrs *lifecycle.PodAdmitAttributes) error {
+	pod := attrs.Pod
+	for _, container := range pod.Spec.InitContainers {
+		if err := h.allocateContainerResources(pod, &container); err != nil {
```

Comment:
Since Init containers start and end before regular containers, devices can be reused.

Since the current behavior already has that behavior maybe this should be the subject of a followup PR / issue.

Reply:
This behavior should be documented though as this is not what is expected

Reply:
Good catch! @jiayingz I think we need a discussion about the reuse of resources  between initContainers and  regular containers, as current allocation is per container.

Reply:
That is a good point. Currently, neither gpuManager nor device plugin supports gpu re-use after init container, so I think it is ok to leave this code as is but adds a TODO comment to address this later.